### DISCUSSION TITLE
Sign client API: Keep-alive using Pings

### DISCUSSION THREAD

**Discussion by nzandrewahn** [WalletConnect/walletconnect-monorepo#4121] [Developer Support]
Currently working on the Wallet side and trying to make sure a connection is still up using the Sign client API but not getting any response from the Dapp when sending a ping. 

```
// for either to ping a peer in a session
  public abstract ping(params: {
    topic: string;
  }): Promise<void>;
```
  
If the DApp hasn't implemented a session handler for the ping event what happens? Will there just be no response if the DApp hasn't implemented a session handler for ping events? How can we test the connection is still alive?

**Comment by glitch-txs** [comment]
hi could you test it here? it should be implemented internally for all dapps.
https://lab.web3modal.com/
